photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,282 @@
+        @property
+        def fitshape(self):
+            return self._fitshape
+        @fitshape.setter
+        def fitshape(self, fitshape):
+            fitshape = np.asarray(fitshape)
+            if len(fitshape) == 2 and np.all(fitshape) > 0:
+                self._fitshape = fitshape
+            else:
+                raise ValueError('fitshape is not defined properly, '
+                                 'received fitshape = {}'.format(fitshape))
+
+    def __call__(self, image):
+        """
+        Parameters
+        ----------
+        image : array-like, ImageHDU, HDUList
+            image to perform photometry
+        
+        Returns
+        -------
+        outtab : astropy.table.Table
+            Table with the photometry results, i.e., centroids and flux
+            estimations.
+        residual_image : array-like, ImageHDU, HDUList
+            Residual image calculated by subtracting the fitted sources
+            and the original image.
+        """
+
+        return self.do_photometry(image)
+
+    def _nstar(self, image, star_groups):
+        """
+        Fit, as appropriate, a compound or single model to a given `groups` of
+        stars. Groups are fitted sequentially from the smallest to the biggest. In
+        each iteration, `image` is subtracted by the previous fitted group. 
+        
+        Parameters
+        ----------
+        image : numpy.ndarray
+            Background-subtracted image.
+        star_groups : list of `~astropy.table.Table`
+            Each `~astropy.table.Table` in this list corresponds to a group of
+            mutually overlapping starts.
+
+        Return
+        ------
+        result_tab : `~astropy.table.Table`
+            Astropy table that contains the results of the photometry.
+        image : numpy.ndarray
+            Residual image.
+        """
+
+        result_tab = Table([[], [], [], [], []],
+                           names=('id', 'group_id', 'x_fit', 'y_fit',
+                                  'flux_fit'),
+                           dtype=('i4', 'i4', 'f8', 'f8', 'f8'))
+
+        star_groups = star_groups.group_by('group_id')
+        
+        groups_order = []
+        for g in star_groups.groups:
+            groups_order.append(len(g))
+
+        N = len(groups_order)
+        while N > 0:
+            curr_order = np.min(groups_order)
+            n = 0
+            N = len(groups_order)
+            while(n < N):
+                if curr_order == len(star_groups.groups[n]):
+                    group_psf = self._get_sum_psf_model(star_groups.groups[n])
+                    x, y, data = self._get_shape_and_data(shape=self.fitshape,\
+                                             star_group=star_groups.groups[n],
+                                                          image=image)
+                    fit_model = self.fitter(group_psf, x, y, data)
+                    print(fit_model)
```

Comment:
I'm to a big fan of unconditional print statements. There is no `verobse=False` flag or similar to switch it off. On the other hand, printing out stuff shows the user that the fit is still going on when the process runs a long time.
What do other people think?